stdlib/http/src/main/ballerina/src/http/auth/utils.bal

```diff
@@ -57,6 +57,29 @@ public function extractAuthorizationHeaderValue(Request req) returns @tainted st
     return req.getHeader(AUTH_HEADER);
 }
 
+# Checks whether the calling resource is secured by evaluating the authentication hierarchy.
+#
+# + context - The `FilterContext` instance
+# + return - The status of calling resource is secured or not
+public function isResourceSecured(FilterContext context) returns boolean {
+    ResourceAuth? resourceLevelAuthAnn = getResourceAuthConfig(context);
+    ServiceAuth? serviceLevelAuthAnn = getServiceAuthConfig(context);
+
+    boolean? resourceSecured = isResourceAuthEnabled(resourceLevelAuthAnn);
+    boolean serviceSecured = isServiceAuthEnabled(serviceLevelAuthAnn);
+
+    if (resourceSecured is boolean) {
+        if (!resourceSecured) {
+            return false;
+        }
+    } else {
+        if (!serviceSecured) {
+            return false;
+        }
+    }
+    return true;
```

Comment:
```suggestion
    if (resourceSecured is boolean) {
        return resourceSecured;
    } else {
        return serviceSecured;
    }
```

Reply:
Oh, I missed this simple logic due to copying and cleaning from exact complex logic. Thanks for pointing out!
https://github.com/ballerina-platform/ballerina-lang/blob/master/stdlib/http/src/main/ballerina/src/http/auth/utils.bal#L73